src/main/java/io/cryostat/configuration/CredentialsManager.java

```diff
@@ -206,22 +190,16 @@ public Credentials getCredentialsByTargetId(String targetId)
         return null;
     }
 
-    public Credentials getCredentials(ServiceRef serviceRef)
-            throws JsonSyntaxException, JsonIOException, IOException, ScriptException {
-        for (String pathString : this.fs.listDirectoryChildren(credentialsDir)) {
-            Path path = credentialsDir.resolve(pathString);
-            try (BufferedReader br = fs.readFile(path)) {
-                StoredCredentials sc = gson.fromJson(br, StoredCredentials.class);
-                if (matchExpressionEvaluator.applies(sc.getMatchExpression(), serviceRef)) {
-                    return sc.getCredentials();
-                }
+    public Credentials getCredentials(ServiceRef serviceRef) throws ScriptException {
+        for (StoredCredentials sc : dao.getAll()) {
```

Comment:
I am wondering if we could convert this to a query in DAO instead of iterating all rows? Not sure if there is any optimization in the database now but we could add an index on this field if it's frequently used?

Reply:
Index on which field? This method tests the stored credentials's match expression against the given ServiceRef. The match expression test can't be done in the database so I think we're pretty much stuck with querying for all rows.

Reply:
oh I mean the `matchExpression` field in the the StoredCredentials table. Something just similar to `deleteByMatchExpression` in DAO but for get?

Reply:
No, it's not that easy unfortunately. `deleteByMatchExpression` deletes a row by String-equality on the match expression column. This method queries by semantic evaluation of the match eexpression, ie the match expression actually gets run as a piece of code.

Reply:
Although now that you point it out, `deleteByMatchExpression` is actually the query we were talking about in the other conversation thread, and it isn't used anywhere because of that `TODO`. I'll remove it. I may as well mark the API handlers as deprecated while I'm at it.

Reply:
ohhh right! Okayyy I see now! Thanks for explaining again :D